lib/dogma/script_sources.ex

```diff
@@ -17,13 +17,23 @@ defmodule Dogma.ScriptSources do
   The `exclude_patterns` argument is a list of regexes. File paths that match
   any of these regexes will not be returned.
   """
-  def find(dir_path, exclude_patterns \\ []) when is_list exclude_patterns do
-    dir_path
-    |> to_string # Convert nil to ""
-    |> Path.join( "**/*.{ex,exs}" )
-    |> Path.wildcard
-    |> Enum.reject( &String.starts_with?(&1, @ignored_dirs) )
-    |> Enum.reject( &match_any?(&1, exclude_patterns) )
+  def find(path, exclude_patterns \\ []) when is_list exclude_patterns do
+    path = to_string path # Convert nil to ""
+    case File.regular? path do
```

Comment:
I don't like all this branching here. Can we extract this into a clearly named function so future readers can understand the intent?